python/ray/worker.py

```diff
@@ -299,7 +299,10 @@ def store_and_register(self, object_id, value, depth=100):
                                        .format(type(e.example_object)))
                     print(warning_message)
                 except (serialization.RayNotDictionarySerializable,
-                        pickle.pickle.PicklingError):
+                        pickle.pickle.PicklingError,
+                        Exception):
```

Comment:
I added this line because now cloudpickle throws a `TypeError` instead of a `PicklingError` when pickling `type(re.compile(r"\d+\.\d*"))`.